src/Tasks.UnitTests/ResourceHandling/GenerateResource_Tests.cs

```diff
@@ -3395,8 +3396,36 @@ public static void AssertLogContainsResource(GenerateResource t, string messageI
         {
             Assert.Contains(
                 String.Format(AssemblyResources.GetString(messageID), replacements),
-                ((MockEngine)t.BuildEngine).Log
-                );
+                ((MockEngine) t.BuildEngine).Log
+            );
+        }
+
+        /// <summary>
+        /// Looks for a formatted message in the output log for the task execution, with unknown formatted parameters.
+        /// If verifies that all constant segments of unformatted message are present.
+        /// </summary>
+        public static void AssertLogContainsResourceWithUnspecifiedReplacements(GenerateResource t, string messageID)
+        {
+            var unformattedMessage = AssemblyResources.GetString(messageID);
+            var matches = Regex.Matches(unformattedMessage, @"\{\d+.*?\}");
+            if (matches.Count > 0)
+            {
+                int i = 0;
+                foreach (Match match in matches)
+                {
+                    string segment = unformattedMessage.Substring(i, match.Index - i);
+                    if (segment.Length > 0)
+                    {
+                        Assert.Contains(segment, ((MockEngine)t.BuildEngine).Log);
+                    }
+
+                    i = match.Index + match.Length;
+                }
```

Comment:
Shouldn't you run `Assert.Contains` on the part after the last placeholder here? Hope I'm not misunderstanding. In general, I would expect to see N+1 asserts where N is the number of placeholders.